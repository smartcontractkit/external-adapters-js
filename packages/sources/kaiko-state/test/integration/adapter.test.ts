import {
  TestAdapter,
  setEnvVariables,
} from '@chainlink/external-adapter-framework/util/testing-utils'
import * as nock from 'nock'
import { MockClientReadableStream } from './fixtures'

describe('execute', () => {
  let spy: jest.SpyInstance
  let testAdapter: TestAdapter
  let oldEnv: NodeJS.ProcessEnv

  beforeAll(async () => {
    oldEnv = JSON.parse(JSON.stringify(process.env))
    process.env.API_KEY = process.env.API_KEY ?? 'fake-api-key'
    process.env.API_ENDPOINT = process.env.API_ENDPOINT ?? 'fake-end-point'
    process.env.BACKGROUND_EXECUTE_MS = process.env.BACKGROUND_EXECUTE_MS ?? '0'
    const mockDate = new Date('2001-01-01T11:11:11.111Z')
    spy = jest.spyOn(Date, 'now').mockReturnValue(mockDate.getTime())

    jest.mock('@kaiko-data/sdk-node/sdk/sdk_grpc_pb', () => ({
      StreamAggregatedStatePriceServiceV1Client: jest.fn().mockImplementation(() => ({
        subscribe: jest.fn(() => new MockClientReadableStream()),
      })),
    }))

    const adapter = (await import('./../../src')).adapter
    adapter.rateLimiting = undefined
    testAdapter = await TestAdapter.startWithMockedCache(adapter, {
      testAdapter: {} as TestAdapter<never>,
    })
  })

  afterAll(async () => {
    setEnvVariables(oldEnv)
    await testAdapter.api.close()
    nock.restore()
    nock.cleanAll()
    spy.mockRestore()
  })

  describe('state endpoint', () => {
    it('should return success', async () => {
      const data = {
        base: 'ETH',
        quote: 'USD',
      }
      const response = await testAdapter.request(data)
      expect(response.statusCode).toBe(200)
      expect(response.json()).toMatchSnapshot()
    })
  })
})
