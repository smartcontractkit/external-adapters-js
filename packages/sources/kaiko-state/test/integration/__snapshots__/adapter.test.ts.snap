// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute state endpoint should return success 1`] = `
{
  "data": {
    "result": 100.2,
  },
  "result": 100.2,
  "statusCode": 200,
  "timestamps": {
    "providerDataReceivedUnixMs": 978347471111,
    "providerDataRequestedUnixMs": 978347471111,
  },
}
`;

